### ISSUE TITLE
Truncated files are cleaned up and then cleaned up again (unsuccessfully)

### ISSUE THREAD

**Issue by bmo-at** [fastify/fastify-multipart#172]
## 🐛 Bug Report

I noticed that in the fix for #160, a slight mistake was introduced: When a file is rejected because of its size, it is immediately cleaned and the errpr gets thrown upwards. Unfortunately, the offending file still makes it into the requests tmpUploads array, where it is then attempted to be cleaned up again when the request draws to a close. There's an easy fix here for which I have already created a pull request: #171

## To Reproduce

```js
const server = require('fastify')({ logger: true });

server.register(require('fastify-multipart'), { limits: { files: 1, fileSize: 1 * 1000 * 1000 } });

server.route({
    method: "POST",
    url: '/upload',
    handler: async (req, res) => {

        let files ;

        try {
            files = await req.saveRequestFiles();
        } catch (error) {
            req.log.error(error);
            res.status(500).send('error');
        }
        if (!res.sent) { res.send(`done. ${files.length} files received`); }
    }
});

server.listen(8080, (error) => {
    if (error) {
        server.log.error(error);
        process.exit(1);
    }
})
```

If you upload a file that is bigger than the set limit, the logger will log an error that 'file could not be deleted'. This is because it has already been deleted.

## Expected behavior

There should be no error thrown for 'file could not be deleted', as it is already deleted.

## Your Environment

- node version: 12
- fastify: 3.4.1
- fastify-multipart: 3.2.0
- os: Debian 10 on WSL2

**Comment by StarpTech** [comment]
Hi @bmo-at thanks for opening an issue.

**Comment by bmo-at** [comment]
Closed by #171
